python/langsmith/client.py

```diff
@@ -4664,6 +4680,543 @@ def _evaluate_strings(
+            else:
+                commit_hash = latest_commit_hash
+
+        response = self.request_with_retries(
+            "GET",
+            (
+                f"/commits/{owner}/{prompt_name}/{commit_hash}"
+                f"{'?include_model=true' if include_model else ''}"
+            ),
+        )
+        return ls_schemas.PromptCommit(
+            **{"owner": owner, "repo": prompt_name, **response.json()}
+        )
+
+    def pull_prompt(
+        self, prompt_identifier: str, *, include_model: Optional[bool] = False
+    ) -> Any:
+        """Pull a prompt and return it as a LangChain PromptTemplate.
+
+        This method requires `langchain_core`.
+
+        Args:
+            prompt_identifier (str): The identifier of the prompt.
+
+        Returns:
+            Any: The prompt object in the specified format.
+        """
+        try:
+            from langchain_core.load.load import loads
+            from langchain_core.prompts import BasePromptTemplate
+            from langchain_core.runnables.base import RunnableSequence
+        except ImportError:
+            raise ImportError(
+                "The client.pull_prompt function requires the langchain_core"
+                "package to run.\nInstall with `pip install langchain_core`"
+            )
+
+        prompt_object = self.pull_prompt_commit(
+            prompt_identifier, include_model=include_model
+        )
+        prompt = loads(json.dumps(prompt_object.manifest))
+
+        if (
+            isinstance(prompt, BasePromptTemplate)
+            or isinstance(prompt, RunnableSequence)
+            and isinstance(prompt.first, BasePromptTemplate)
+        ):
+            prompt_template = (
+                prompt
+                if isinstance(prompt, BasePromptTemplate)
+                else (
+                    prompt.first
+                    if isinstance(prompt, RunnableSequence)
+                    and isinstance(prompt.first, BasePromptTemplate)
+                    else None
+                )
+            )
+            if prompt_template is None:
+                raise ls_utils.LangSmithError(
+                    "Prompt object is not a valid prompt template."
+                )
+
+            if prompt_template.metadata is None:
+                prompt_template.metadata = {}
+            prompt_template.metadata.update(
+                {
+                    "lc_hub_owner": prompt_object.owner,
+                    "lc_hub_repo": prompt_object.repo,
+                    "lc_hub_commit_hash": prompt_object.commit_hash,
+                }
+            )
+
+        return prompt
+
+    def push_prompt(
+        self,
+        prompt_identifier: str,
+        *,
+        object: Optional[Any] = None,
+        parent_commit_hash: Optional[str] = "latest",
+        is_public: bool = False,
+        description: Optional[str] = "",
+        readme: Optional[str] = "",
+        tags: Optional[List[str]] = [],
```

Comment:
We shouldn't default to a list (a mutable type) in the type signature. Should initialize the list within the function itself